enterprise/admin/views.py

```diff
@@ -102,6 +104,32 @@ def get_user_name(request):
         return request.user.first_name or request.user.username
 
 
+class CatalogQueryPreviewView(View):
+    """
+    Renders a search/all response from Discovery for a particular catalog query.
+    """
+
+    def get(self, request, catalog_query_id):
+        """
+        Render the response for a particular catalog query.
+        """
+        enterprise_catalog_query = get_object_or_404(EnterpriseCatalogQuery, pk=catalog_query_id)
+        if not enterprise_catalog_query:
+            return HttpResponse(status=404)
+        return JsonResponse(self.get_response_from_discovery(request, enterprise_catalog_query))
+
+    @staticmethod
+    def get_response_from_discovery(request, catalog_query):
+        """
+        POST a query to Discovery's search/all endpoint and return the results.
+        """
+        response = get_course_catalog_api_service_client().get_catalog_results(
```

Comment:
Please pass the `exclude_expired` and `detail_fields` in the `query_params` as this might create an issue with the results of search/all endpoint because we get these parameters from the GET request in the course serializer of search/all API endpoint. For reference of getting parameters from GET request https://github.com/bilaltahir21/course-discovery/blob/e2ea41d45eccd023c13cd78cf36f313046119fc5/course_discovery/apps/course_metadata/search_indexes/serializers/course.py#L67

Reply:
Yes, I think we should see if any of these keys are present in `catalog_query.content_filter` then we should extract them from `catalog_query.content_filter` and pass them in `query_params` of the `get_catalog_results` method.

Reply:
Do we ever use these in the content_filter?

Reply:
the GET preview did not apply this processing on the content_filter ...

Reply:
I would avoid modifying the clients to fix an issue that is best handled on the server side
 … from an architecture perspective the client side should not have to worry about which filters to send and which not …